test/functional/test_framework/test_framework.py

```diff
@@ -227,6 +229,11 @@ def setup(self):
             if not self.supports_cli:
                 raise SkipTest("--usecli specified but test does not support using CLI")
             self.skip_if_no_cli()
+
+        #Diasble timeout if flag is set
+        if self.options.notimeout:
```

Comment:
Should it warn, raise, or not be set if `--factor` is passed?

Reply:
I added a message in the help text. But yes, it would be a good idea to enforce it.

Reply:
I'm thinking either --notimeout should override any --factor setting and output a warning if it does, or halt with an explanation. Depending on which, the helps should be updated. Curious what others think.

Reply:
I think it's better to immediately raise an error stating these two should not be set together. 
Generally, someone wanting to disable timeout will not have a need to set --factor. Catching that would be sufficient.
But there can be an edge case where for some reason they need --factor but also want to attach gdb to the process. Right now that's not possible without much deeper refactoring. currently http responses in a test to node, and node to node, are checked in the same function . So it's not possible to only disable it for test to node, but keep it for node to node.